pwndbg/commands/xinfo.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import gdb
+import subprocess
+
+import pwndbg.arch
+import pwndbg.commands
+import pwndbg.config
+import pwndbg.memory
+import pwndbg.vmmap
+import pwndbg.regs
+import pwndbg.color.memory as M
+
+
+parser = argparse.ArgumentParser(description='Shows offsets of the specified address to useful other locations')
+parser.add_argument('address', nargs='?', default='$pc', 
+                    help='Address to inspect')
+
+def print_line(name, addr, first, second, op, width = 20):
+    FMT = "{} {} = {} {} {:#x}"
+
+    print(FMT.format(
+        name.rjust(width),
+        M.get(addr),
+        M.get(first) if type(first) is not str else first.ljust(len(hex(addr))),
+        op,
+        second,))
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def xinfo(address=None):
+    addr = int(address)
+    addr &= pwndbg.arch.ptrmask
+
+    page = pwndbg.vmmap.find(addr)
+
+    print("Extended information for virtual address {}:".format(M.get(addr)))
+
+    if page is None:
+        print("\n  Virtual address {:#x} is not mapped.".format(addr))
+        return
+
+    print("\n  Containing mapping:")
+    print(M.get(address, text=str(page)))
+
+    print("\n  Offset information:")
+
+    if page.objfile == "[stack]":
+        # If it's a stack address, print offsets to top and bottom of stack, as
+        # well as offsets to current stack and base pointer (if used by debugee)
+        sp = pwndbg.regs.sp
+        frame = pwndbg.regs[pwndbg.regs.frame]
+        frame_mapping = pwndbg.vmmap.find(frame)
+
+        print_line("Stack Top", addr, page.vaddr, addr - page.vaddr, "+")
+        print_line("Stack End", addr, page.vaddr + page.memsz, page.memsz -
+            (addr - page.vaddr), "-")
+        print_line("Stack Pointer", addr, sp, addr - sp, "+")
+
+        if frame_mapping and page.vaddr == frame_mapping.vaddr:
+            print_line("Frame Pointer", addr, frame, frame - addr, "-")
+
+    else:
+        # Otherwise, just print the distance to the beginning of the mapping
+        print_line("Mapped Area", addr, page.vaddr, addr - page.vaddr, "+")
+
+    if len(page.objfile) > 0 and page.objfile[0] != "[":
+        # If it's an address pointing into a memory mapped file, print offsets
+        # to beginning of file in memory and on disk
+        file_name = page.objfile
+        objpages = filter(lambda p: p.objfile == file_name, pwndbg.vmmap.get())
+        objpages = sorted(objpages, key = lambda p: p.vaddr)
+        first = objpages[0]
+        rva = addr - first.vaddr
+
+        print_line("File (Memory)", addr, first.vaddr, rva, "+")
+
+        p = subprocess.Popen(["readelf", "-l", file_name], stdout = subprocess.PIPE)
```

Comment:
Since calling other processes is not that easy topic (what with other achitectures? what if the binary is not present? and so on) we create wrappers for commands. Please add appropriate function to https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/wrappers/readelf.py and use it here instead of calling it directly.

NOTE: Thanks to this we can try to use different readelf names for different architectures (for remote debugging, when binary is present locally) and so on (this isn't yet there, but lets have a possibility to add it easily).